pygeos/io.py

```diff
@@ -219,33 +296,47 @@ def from_shapely(geometry, **kwargs):
     >>> from shapely.geometry import Point   # doctest: +SKIP
     >>> from_shapely(Point(1, 2))   # doctest: +SKIP
     <pygeos.Geometry POINT (1 2)>
+
+    Notes
+    -----
+    If PyGEOS and Shapely do not use the same GEOS version,
+    the conversion happens through the WKB format and will thus be slower.
     """
     check_shapely_version()
-
-    if shapely_geos_version is None:
+    if shapely_compatible is None:
         raise ImportError("This function requires shapely")
 
-    # shapely has something like: "3.6.2-CAPI-1.10.2 4d2925d6"
-    # pygeos has something like: "3.6.2-CAPI-1.10.2"
-    if not shapely_geos_version.startswith(geos_capi_version_string):
-        raise ImportError(
-            "The shapely GEOS version ({}) is incompatible with the GEOS "
-            "version PyGEOS was compiled with ({})".format(
-                shapely_geos_version, geos_capi_version_string
-            )
-        )
-
-    if isinstance(geometry, ShapelyGeometry):
-        # this so that the __array_interface__ of the shapely geometry is not
-        # used, converting the Geometry to its coordinates
-        arr = np.empty(1, dtype=object)
-        arr[0] = geometry
-        arr.shape = ()
-    elif not isinstance(geometry, np.ndarray) and isinstance(geometry, Sized):
-        # geometry is a list/array-like
-        arr = np.empty(len(geometry), dtype=object)
-        arr[:] = geometry
+    if shapely_compatible:
+        if isinstance(geometry, ShapelyGeometry):
+            # this so that the __array_interface__ of the shapely geometry is not
+            # used, converting the Geometry to its coordinates
+            arr = np.empty(1, dtype=object)
+            arr[0] = geometry
+            arr.shape = ()
+        elif not isinstance(geometry, np.ndarray) and isinstance(geometry, Sized):
+            # geometry is a list/array-like
+            arr = np.empty(len(geometry), dtype=object)
+            arr[:] = geometry
+        else:
+            # we already have a numpy array or we are None
+            arr = geometry
+
+        return lib.from_shapely(arr, **kwargs)
     else:
-        # we already have a numpy array or we are None
-        arr = geometry
-    return lib.from_shapely(arr, **kwargs)
+        unpack = geometry is None or isinstance(geometry, ShapelyGeometry)
+        if unpack:
+            geometry = (geometry,)
+
+        arr = [
+            None
+            if g is None
```

Comment:
This double `if ... else`  in a list comprehension is a bit too long for my taste. 
Also I think that the "empty point" exception is so rare that this should be a case of "exception handling" instead of preemptive checking.

Reply:
I can rewrite this part as a normal for loop. It will probably take a small performance hit as we would need to append to the list, and python cannot pre-allocate as it does with list comprehensions, but I don't think it will be that noticeable.

I tried using a try/except, but shapely somehow throws an `IllegalArgumentException` that I cannot catch:
``` python
>>> import shapely
>>> from shapely import wkt
>>> empty = wkt.loads('POINT EMPTY')
>>> try:
...   empty.wkb
... except:
...   print('caught exception')
IllegalArgumentException: Empty Points cannot be represented in WKB
```

Reply:
I just rewrote this to a normal loop, as I felt it was necessary in order to work with prepared geometries.